code/src/cljs/sixsq/nuvla/ui/pages/edges_detail/views_coe_resources.cljs

```diff
@@ -155,30 +155,94 @@
+               registries                   (subscribe [::subs/registries])
+               set-use-private-registry?    #(swap! !image-spec assoc :use-private-registry? %)
+               set-private-registry-id      (fn [id]
+                                              (let [{:keys [endpoint] :as _private-registry} (first (filter #(= id (:id %)) @registries))
+                                                    registry-endpoint (some-> endpoint
+                                                                              (cond-> (not (str/ends-with? endpoint "/"))
+                                                                                      (str "/")))]
+                                                (swap! !image-spec assoc
+                                                       :private-registry-id id
+                                                       :private-registry-endpoint registry-endpoint)))
+               set-private-registry-cred-id #(swap! !image-spec assoc :private-registry-cred-id %)
+               set-image-name               #(swap! !image-spec assoc :image %)]
+    (let [{:keys [image use-private-registry? private-registry-id private-registry-endpoint private-registry-cred-id]} @!image-spec
+          registries-options      (subscribe [::subs/registries-options])
+          registries-cred-options (subscribe [::subs/registries-credentials-options private-registry-id])]
+      [ui/Form
+       (when-not (empty? @registries)
+         [:<>
+          [ui/FormCheckbox {:label     (@tr [:private-registry])
+                            :checked   use-private-registry?
+                            :on-change (ui-callback/checked set-use-private-registry?)}]
+          (when use-private-registry?
+            [ui/Table {:style {:margin-top 10}}
+             [ui/TableHeader
+              [ui/TableRow
+               [ui/TableHeaderCell {:content (r/as-element
+                                               [:<>
+                                                (@tr [:private-registry])
+                                                [:span {:style {:color "red" :margin-left "4px"}} "*"]])}]
+               [ui/TableHeaderCell {:content (str/capitalize (@tr [:credential]))}]]]
+             [ui/TableBody
+              [ui/TableRow
+               [ui/TableCell {:width 7}
+                [ui/Dropdown
+                 {:selection true
+                  :fluid     true
+                  :value     private-registry-id
+                  :options   @registries-options
+                  :on-change (ui-callback/value set-private-registry-id)}]]
+               [ui/TableCell {:width 7}
+                [ui/Dropdown
+                 {:selection true
+                  :fluid     true
+                  :value     private-registry-cred-id
+                  :options   @registries-cred-options
+                  :on-change (ui-callback/value set-private-registry-cred-id)}]]]]])])
```

Comment:
I think we may have a bug if the user do the following:
e.g.
1 . select a registry a
2. select a cred for registry a
3. select registry b and call pull image
In this case the call will be done with the cred of registry a